backend/tests/test_logging_axiom.py

```
@@ -0,0 +1,564 @@
+            # Drive a single flush directly — no exception should escape.
+            handler._worker._flush([{"_time": "now", "message": "hi"}])
+
+            captured = capsys.readouterr()
+            assert "[axiom-logger]" in captured.err
+            assert "ingest transport error" in captured.err
+            assert "ConnectError" in captured.err
+        finally:
+            _stop_handler(handler)
+
+    def test_logger_remains_functional_after_axiom_failure(
+        self, isolated_root_logger, mock_axiom_token, capsys
```

Ruff (0.15.13)

[warning] Unused method argument: `isolated_root_logger`

(ARG002)